dora/core/server/master/src/main/java/alluxio/master/scheduler/Scheduler.java

```diff
@@ -527,17 +527,18 @@ public void kickStartTasks() {
             tasksQ.size());
         CloseableResource<BlockWorkerClient> blkWorkerClientResource
             = mActiveWorkers.get(workerInfo);
-        if (blkWorkerClientResource == null) {
-          LOG.warn("Didn't find corresponding BlockWorkerClient for workerInfo:{}",
-              workerInfo);
-          return;
-        }
         Task task = tasksQ.peek();
         // only make sure 1 task is running at the time
         if (task == null || task.getResponseFuture() != null) {
           LOG.debug("head task is {}", (task == null) ? "NULL" : "already running");
           return;
         }
+        if (blkWorkerClientResource == null) {
```

Comment:
I'm not sure about this either.
If the change above is applied and when USER_DYNAMIC_CONSISTENT_HASH_RING_ENABLED  is set true, membership change will lead to a subtask to be dropped. We need a way to notify the job so that it can fail the task.

Reply:
But I guess here if we add file directly to failure is also not a correct behavior under dynamic hash ring situation since we want to assign task to another worker.

Reply:
my bad. i meant add this file to retry instead of just failing it.